### PR TITLE
Added resource specification for pods in LightGBM API

### PR DESCRIPTION
**Pull request by karthikv2k**
Summary:
1. Added resource specification (CPU, and Memory) for pods in LightGBM API
1. Added ability to specify pod spec mutators to serve as an escape hatch for the functionalities not exposed by the API
1. Increased nslookup retires to adapt to delays in node provisioning in auto scaled node pools
1. Added unit tests
1. Added crcmod to Dockerfile to improve download speed in gsutil

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/fairing/262)

### CODE REVIEW

File: examples/lightgbm/distributed-training.ipynb [hunk 1]
```diff
@@ -81,7 +81,11 @@
    },
    "outputs": [],
    "source": [
-    "lightgbm.execute(config=params, docker_registry=DOCKER_REGISTRY)"
+    "lightgbm.execute(config=params,\n",
+    "                          docker_registry=DOCKER_REGISTRY,\n",
+    "                          cores_per_worker=1,\n",
```
Comment (abhi-g) [inline examples/lightgbm/distributed-training.ipynb]: Does cores_per_worker specify #cores assigned to each worker or to allocate a single core node?

File: examples/lightgbm/distributed-training.ipynb [hunk 2]
```diff
@@ -81,7 +81,11 @@
...
+    "                          memory_per_worker=0.5,\n",
```
Comment (abhi-g) [inline examples/lightgbm/distributed-training.ipynb]: similarly, is this 1/2 the ram of the node, or 0.5GB? possibly adding a comment at the end of the line might clarify the usage. something like:
 memory_per_worker=0.5,\n",   # reserve 0.5GB for each worker

File: fairing/frameworks/lightgbm.py [hunk 3]
```diff
@@ -149,6 +154,12 @@ def execute(config,
         docker_registry: registry to push the built docker image
         base_image: base image to use for this job. It should have lightgbm installed and should be in PATH variable.
         namespace: Kubernetes namespace to use
+        stream_log: True - streams logs from the first worker in the training job after job launch till the training is finished.
```
Comment (abhi-g) [inline fairing/frameworks/lightgbm.py]: Thanks for documenting these. This should show up in help(lightgbm.execute), correct?

File: fairing/frameworks/utils.py [hunk 4]
```diff
@@ -30,7 +30,14 @@ def parse_cluster_spec_env():
     return hosts, ips, ports
 
 
-def nslookup(hostname, retries=20):
+def nslookup(hostname, retries=600):
+    """
+    Does nslookup for the hostname and returns the IPs for it.
+    Attributes:
+        hostname: hostname to be looked up
+        retries: #retries before failing. In autoscalled cluster, it might take upto 10mins
```
Comment (abhi-g) [inline fairing/frameworks/utils.py]: s/autoscalled/autoscaled/

File: examples/lightgbm/distributed-training.ipynb (hunk 1 above)
Comment (karthikv2k) [reply examples/lightgbm/distributed-training.ipynb]: #cores assigned to each worker . Added doc strings.

File: examples/lightgbm/distributed-training.ipynb (hunk 2 above)
Comment (karthikv2k) [reply examples/lightgbm/distributed-training.ipynb]: 0.5gb . Added doc strings.

File: fairing/frameworks/lightgbm.py (hunk 3 above)
Comment (karthikv2k) [reply fairing/frameworks/lightgbm.py]: yes.

File: fairing/frameworks/utils.py (hunk 4 above)
Comment (karthikv2k) [reply fairing/frameworks/utils.py]: done.

**Pull request merged**